packages/thirdweb/src/contract/deployment/zksync/zkDeployCreate2Factory.ts

```diff
@@ -49,25 +33,52 @@ export async function zkDeployCreate2Factory(
   });
 
   if (balance.value < valueToSend) {
-    await sendAndConfirmTransaction({
-      account: options.account,
-      transaction: prepareTransaction({
-        chain: options.chain,
-        client: options.client,
-        to: create2Signer.address,
-        value: valueToSend,
-      }),
+    return prepareTransaction({
```

Comment:
this tx should be done before the second one, so ideally we return 2 transactions in an array here? and we execute them one by one